microscopium/serve.py

```diff
@@ -267,6 +270,19 @@ def update_table(indices, df, table):
     filtered_df = df.iloc[indices]
     table.source.data = ColumnDataSource(filtered_df).data
 
+def switch_modes_button_group():
+    from bokeh.models.widgets import RadioButtonGroup
+
+    radio_button_group = RadioButtonGroup(
+        labels=["tSNE", "UMAP", "PCA"], active=0)
+    return radio_button_group
+
+def update_plot(my_plot, source, button_dict, mode, glyph_size=1, alpha_value=0.8):
+    x_source = "x." + button_dict[mode]
+    y_source = "y." + button_dict[mode]
+    source.data['x'] = source.data[x_source]
```

Comment:
Very nice!
I don't think this will rescale the axis of the scatterplot figure, which is something we'll likely need to do too. Depending on the dataset, it can be pretty common to have entirely different ranges from the different clustering methods.

Reply:
Tuns out this is easy enough to set, so I've done that with:
```python
embed.x_range.range_padding
embed.x_range.range_padding_units
```
See this stackoverflow answer for more details: https://stackoverflow.com/questions/34929370/python-bokeh-modify-axis-scale
And the bokeh docs: http://bokeh.pydata.org/en/latest/docs/reference/models/ranges.html#bokeh.models.ranges.DataRange1d.range_padding